packages/react/src/components/Comments/Thread.tsx

```diff
@@ -118,38 +76,111 @@ export const Thread = ({
     newCommentEditor.removeBlocks(newCommentEditor.document);
   }, [comments, newCommentEditor, thread.id]);
 
-  showComposer =
-    showComposer && comments.threadStore.auth.canAddComment(thread);
+  const [showAllComments, setShowAllComments] = useState(
+    view === "floating" ||
+      thread.comments.length <= (maxCommentsBeforeCollapse || 5)
+  );
+
+  const commentElements = useMemo(() => {
+    if (!showAllComments) {
+      return [
+        <Comment
+          key={thread.comments[0].id}
+          thread={thread}
+          comment={thread.comments[0]}
+          index={0}
+        />,
+        <Components.Comments.ExpandSectionsPrompt
+          key={"expand-prompt"}
+          className={"bn-thread-expand-prompt"}
+          onClick={(event) => {
+            setShowAllComments(true);
+            editor.comments?.selectThread(thread.id);
+            (
+              (event.target as Element).closest(
+                ".bn-thread"
+              ) as HTMLElement | null
+            )?.focus();
+          }}>
+          {`${thread.comments.length - 2} more replies`}
+        </Components.Comments.ExpandSectionsPrompt>,
+        <Comment
+          key={thread.comments[thread.comments.length - 1].id}
+          thread={thread}
+          comment={thread.comments[thread.comments.length - 1]}
+          index={thread.comments.length - 1}
+        />,
+      ];
+    }
+
+    return thread.comments.map((comment, index) => {
+      return (
+        <Comment
+          key={comment.id}
+          thread={thread}
+          comment={comment}
+          index={index}
+        />
+      );
+    });
+  }, [Components, editor.comments, showAllComments, thread]);
 
   return (
     <Components.Comments.Card
-      className={mergeCSSClasses("bn-thread", className)}
-      {...props}>
-      <Components.Comments.CardSection className="bn-thread-comments">
-        {thread.comments.map((comment, index) => {
-          const isFirstComment = index === firstCommentIndex;
-
-          return (
-            <Comment
-              key={comment.id}
-              thread={thread}
-              className="bn-thread-comment"
-              comment={comment}
-              showDeleted={showDeletedComments}
-              showActions={showActions}
-              showReactions={showReactions}
-              showResolveAction={isFirstComment}
-            />
+      className={"bn-thread"}
+      headerText={highlightedText}
+      onFocus={() => {
+        if (view === "floating") {
+          return;
+        }
+
+        comments.selectThread(thread.id);
+      }}
+      onBlur={(event) => {
+        if (view === "floating") {
+          return;
+        }
+
+        if (
+          !(event.relatedTarget instanceof Node) ||
+          !(event.target instanceof Node) ||
+          !event.target.contains(event.relatedTarget)
+        ) {
+          setShowAllComments(
+            thread.comments.length <= (maxCommentsBeforeCollapse || 5)
```

Comment:
I don't think this should be based on blur, but based on the state of whether a Thread is selected.

i.e.: when selecting a thread in the editor, it should expand, and when unselecting, it should collapse (didn't test this yet, but judging from the code)